Cachix documentation and agent config reference (#25)
hercules-ci/docs.hercules-ci.com: docs/agent-config.adoc

> = Agent configuration

Reviewer: Observation: this section needs to duplicate a lot of what's also in the module documentation. I estimate that DRY is not feasible.

> Agent can be configured by running `hercules-ci-agent --config FILE.toml`

Reviewer: ```suggestion
Agent can be run with a configuration by invoking `hercules-ci-agent --config FILE.toml`.
```

> config formatted using https://github.com/toml-lang/toml#toml[TOML] markup.

Reviewer: We recommend to use one of the declarative deployment methods, particularly because those also configure Nix.

> 

Reviewer: A minimal example file would be nice, to demonstrate the syntax and as a starting point.

== apiBaseUrl

Optional. Defaults to https://hercules-ci.com.

HTTP API agent will connect to.

== clusterJoinTokenPath

Required.

Path to a secret token retrieved when creating a new agent via
https://hercules-ci.com for HTTP API authentication.

== concurrentTasks

> Optional. Defaults to number of cores.

Reviewer: Current situation is different. I'm not in a hurry to change this.
```suggestion
Optional. Defaults to 4.
```

Number of workers to use for building/evaluating Nix derivations.

== binaryCachesPath

Optional. No default.

Path to a JSON file containing binary cache secret keys.

Currently only support http://cachix.org/[Cachix] binary caches.

Example:

[source,json]
----
{ "my-private":
    { "kind": "CachixCache",

>       "authToken": "af2353653abc23"

Reviewer: ```suggestion
      "authToken": "eyJhaf23GH53a.bc23BUSI.9q3048hWHh"
```

>     , "publicKeys": ["dummy.cachix.org-1:EjBSHzF6VmDnzqlldGXbi0RM3HdjfTU3yDRi9Pd0jTY="]

Reviewer: ```suggestion
    , "publicKeys": ["my-private.cachix.org-1:EjBSHzF6VmDnzqlldGXbi0RM3HdjfTU3yDRi9Pd0jTY="]
```

> = Agent configuration

Reviewer: ```suggestion
= Agent configuration file
```

Assuming the purpose of this section is to serve as a reference.

> Agent can be run with a configuration by invoking `hercules-ci-agent --config FILE.toml`.

Reviewer: ```suggestion
The agent can be run with a configuration by invoking `hercules-ci-agent --config agent.toml`.
```

"The" + modify the filename to match the suggested filename at the example. No need to use a metavariable.

config formatted using https://github.com/toml-lang/toml#toml[TOML] markup.

> By default you don't need to specify the config file. A bit more involved example:

Reviewer: ```suggestion
When deploying with NixOS, NixOps or nix-darwin, you don't need to specify the config file.
```

> 

Reviewer: ```suggestion

A basic `agent.toml` looks as follows:

```

Suggesting a default file name too.

[source,toml]
----

> apiBaseUrl = "https://hercules-ci.com"

Reviewer: ```suggestion
```

The example serves as an introduction and doesn't have to be complete.
